apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -175,29 +167,16 @@ private async Task<PooledSession> AcquireSessionImplAsync(TransactionOptions tra
                 // We do this when a session is released, and in the maintenance task.
                 // These happen frequently enough that we shouldn't need to worry about them here.
 
-                // Update statistics for sessions with read/write transaction options - only after we've already acquired the session.
-                if (transactionOptions.ModeCase == ModeOneofCase.ReadWrite)
-                {
-                    Interlocked.Increment(ref _rwTransactionRequests);
-                    if (session.TransactionOptions.ModeCase == transactionMode)
-                    {
-                        Interlocked.Increment(ref _rwTransactionRequestsPrewarmed);
-                    }
-                }
-
-                // Whatever session we got it either has vanilla transaction options or the wrong ones.
-                return session.WithTransactionOptions(transactionOptions, singleUseTransaction);
+                // While in the pool, sessions have TransactionOptions.ModeCase == ModeOneofCase.None and singleUseTransaction == false.
```

Comment:
This makes me wonder whether we've got the model quite right... whether we should consider refactoring to "sessions idle in the pool" and "sessions from the pool, now being used" as different types. But let's chat about that.